microservices/s3_to_redshift/s3_to_redshift.py

```diff
@@ -139,9 +138,34 @@ def copy_query(dbtable, batchfile, log):
 
         obj = client.get_object(Bucket=bucket, Key=object_summary.key)
         body = obj['Body']
+        # Create an object to hold the data while parsing
+        csv_string = ''
+        # Perform regex pattern replacements according to config, if defined
+        if 'global_regex' in data:
+            inline_pattern = data['global_regex']['string_repl']['pattern']
+            inline_replace = data['global_regex']['string_repl']['replace']
```

Comment:
if `data['global_regex']['string_repl']` is not defined in configuration _or_ if _either_ of `['pattern']` or `['replace']` weren't defined therein, then these assignments will result in a `KeyError`.

Reply:
you either need to handle for this, or not let it happen in the first place by wrapping these in an `if(data['global_regex']['string_repl']):` block

Reply:
I have added a check to see if string_repl is defined in the config.